ktor-network/ktor-network-tls/ktor-network-tls-certificates/jvm/src/io/ktor/network/tls/certificates/builders.kt

```diff
@@ -101,9 +161,10 @@ public class KeyStoreBuilder internal constructor() {
         store.load(null, null)
 
         certificates.forEach { (alias, info) ->
-            val (certificate, keys, password) = info
-            store.setCertificateEntry(alias, certificate)
-            store.setKeyEntry(alias, keys.private, password.toCharArray(), arrayOf(certificate))
+            val (certificate, keys, password, issuerCertificate) = info
+            store.setCertificateEntry(alias + "Cert", certificate)
```

Comment:
A note on this change. `setCertificateEntry` uses the same hashtable internally as `setKeyEntry`, which means calling `setKeyEntry` with the same alias makes `setCertificateEntry` redundant - we could remove this line entirely.

However, one of the `generateCertificate` functions adds the certificate to the keystore using this `Cert` suffix, so to ensure backwards compatibility with existing usages of `generateCertificate`, I added this suffix here too instead of removing the line. If you believe it's not worth keeping it backwards compatible to this extent, we can remove the line and the corresponding test.

Reply:
It turns out the tests are failing because this change is backwards-incompatible for the other code paths: some tests iterate on all aliases in the keystore and they expect to get a certificate chain out of every alias, which only works for aliases on private keys, not for aliases of plain certificates.

I will remove this line entirely here, and only add the certificate under the `<keyAlias>Cert` alias in the case of the `generateCertificate` function that used to do it. It should be possible by accessing the generated certificate from the key store and then re-adding it under the other alias.

Reply:
Done.